File: ruby/supplies_package.rb
module Deepspace
  class SuppliesPackage
    def initialize(a, f, s)
      @ammoPower = a
      @fuelUnits = f
      @shieldPower = s
    end

    def self.newCopy(otro)
      s = otro.clone
    end

    def getAmmoPower
      @ammoPower
    end

    def getFuelUnits
      @fuelUnits
    end

    def getShieldPower
      @shieldPower
    end
  end
end

File: java/PracticaPDOO/src/main/java/deepspace/SuppliesPackage.java
/**
 *
 * @author Alejandro Villanueva Prados
 */
package deepspace;

public class SuppliesPackage {
   private float ammoPower;
   private float fuelUnits;
   private float shieldPower;
   
   SuppliesPackage(float a, float f, float s) {
       ammoPower = a;
       fuelUnits = f;
       shieldPower = s;
   }
   
   SuppliesPackage(SuppliesPackage otro) {
       ammoPower = otro.ammoPower;
       fuelUnits = otro.fuelUnits;
       shieldPower = otro.shieldPower;
   }
   
   float getAmmoPower() {
       return ammoPower;
   }
   
   float getFuelUnits() {
       return fuelUnits;
   }
   
   float getShieldPower() {
       return shieldPower;
   }
   
   /* Gracias a esta sobrecarga, obtenemos mucha información sobre el objeto, 
    sobre todo para comparar que estamos realizando la copia correctamente */
   @Override
   public String toString() {
       return "SuppliesPackage[" + "@" + System.identityHashCode(this) + getAmmoPower() + ", " + getFuelUnits() + ", " + getShieldPower() + "]";
   }
}
